core/commands/repo.go

```diff
@@ -259,7 +263,103 @@ daemons are running.
 	},
 	Type: MessageOutput{},
 	Marshalers: oldcmds.MarshalerMap{
-		oldcmds.Text: MessageTextMarshaler,
+		cmds.Text: MessageTextMarshaler,
+	},
+}
+
+var repoRmRootCmd = &cmds.Command{
+	Helptext: cmdkit.HelpText{
+		Tagline: "Unlink the root used by the files API.",
+		ShortDescription: `
+'ipfs repo rm-root' will unlink the root used by the files API ('ipfs
+files' commands) without trying to read the root itself.  The root and
+its children will be removed the next time the garbage collector runs,
+unless pinned.
+
+This command is designed to recover form the situation when the root
+becomes unavailable and recovering it (such as recreating it, or
+fetching it from the network) is not possible.  This command should
+only be used as a last resort as using this command could lead to data
+loss if there are unpinned nodes connected to the root.
+
+This command can only run when the ipfs daemon is not running.
+`,
+	},
+	Options: []cmdkit.Option{
+		cmdkit.BoolOption("confirm", "Really perform operation."),
+		cmdkit.BoolOption("remove-local-root", "Remove even if the root exists locally."),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) {
+		confirm, _ := req.Options["confirm"].(bool)
+		removeLocalRoot, _ := req.Options["remove-local-root"].(bool)
+
+		if !confirm {
+			res.SetError(fmt.Errorf("this is a potentially dangerous operation please pass --confirm to proceed"), cmdkit.ErrNormal)
+			return
+		}
+
+		ctx, ok := env.(*oldcmds.Context)
+		if !ok {
+			res.SetError(fmt.Errorf("expected env to be of type %T, got %T", ctx, env), cmdkit.ErrNormal)
+			return
+		}
+		configRoot := ctx.ConfigRoot
+
+		// Can't use a full node as that interferes with the removal
+		// of the files root, so open the repo directly
+		repo, err := fsrepo.Open(configRoot)
+		if err != nil {
+			res.SetError(err, cmdkit.ErrNormal)
+			return
+		}
+		defer repo.Close()
+		bs := bstore.NewBlockstore(repo.Datastore())
+
+		// Get the old root and display it to the user so that they can
+		// can do something to prevent from being garbage collected,
+		// such as pin it
+		dsk := core.FilesRootKey()
+		val, err := repo.Datastore().Get(dsk)
+		if err == ds.ErrNotFound || val == nil {
+			cmds.EmitOnce(res, &MessageOutput{"Files API root not found.\n"})
+			return
+		}
+
+		var cidStr string
+		var have bool
+		c, err := cid.Cast(val.([]byte))
+		if err == nil {
+			cidStr = c.String()
+			have, _ = bs.Has(c)
+		} else {
+			cidStr = b58.Encode(val.([]byte))
+		}
+
+		if have && !removeLocalRoot {
+			res.SetError(fmt.Errorf("root %s exists locally. Are you sure you want to unlink this? Pass --remove-local-root to continue", cidStr), cmdkit.ErrNormal)
+			return
+		} else if !have && removeLocalRoot {
+			res.SetError(fmt.Errorf("root does not %s exists locally. Please remove --remove-local-root to continue", cidStr), cmdkit.ErrNormal)
```

Comment:
The documentation says "even if" not "only if" so I'd expect this case to work.

Reply:
I can fix the documentation.  The idea behind this is to prevent this flag from automatically being passed.  It is intentionally annoying.  if this will Create a problem I can remove this check.

Reply:
Reviving...

IMO, having to pass `--confirm` is annoying enough. If a user wants to automate this, that's their problem (and their choice).

I've switched this to act as documented unless someone objects.